File: NetFramework/FlexGrid/VB/ColumnBandsOrderManagment/ColumnBandsUserControl.vb
﻿Imports System.Data.OleDb
Imports C1.Win.C1Themes
Imports C1.Win.C1FlexGrid

Public Enum LayoutOption
    Traditional
    Compact
    Detailed
End Enum

Partial Public Class ColumnBandsUserControl
    Inherits UserControl

    Public Sub New()
        InitializeComponent()
    End Sub

    Public Sub Init()
        SetupC1FlexGrid()
    End Sub

    Public Sub ApplyTheme(ByVal themeName As String)
        Dim theme = C1ThemeController.GetThemeByName(themeName, False)
        C1ThemeController.ApplyThemeToControlTree(Me, theme)
    End Sub

    Private Sub SetupC1FlexGrid()
        flexGrid1.BeginUpdate()
        flexGrid1.DataSource = GetDataSource()
        flexGrid1.AutoSizeCols()
        flexGrid1.AutoSizeRows()
        flexGrid1.GroupDescriptions = New List(Of GroupDescription)() From {New GroupDescription("CompanyName")}
        flexGrid1.Cols(1).Width = 300
        flexGrid1.EndUpdate()
    End Sub

    Private Function GetDataSource() As DataTable
        Dim rs As String = "
                SELECT Distinct 
	                Orders.[OrderID],
                    Products.ProductID,
	                Products.ProductName, 
	                Orders.OrderDate, 
                    Orders.ShippedDate as DeliveryDate,
                    Orders.ShipCountry,
                    Orders.ShipCity,
                    Orders.ShipAddress,
	                Customers.CompanyName, 
	                [Order Details].UnitPrice AS UnitPrice, 
	                [Order Details].Quantity, 
	                [Order Details].Discount, 
	                ([Order Details].[UnitPrice]*[Quantity]*(1-[Discount])/100)*100 as Amount 
	                FROM Shippers 
		                INNER JOIN (Products 
		                INNER JOIN ((Employees 
		                INNER JOIN (Customers 
		                INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID) ON Employees.EmployeeID = Orders.EmployeeID) 
		                INNER JOIN [Order Details] ON Orders.OrderID = [Order Details].OrderID) ON Products.ProductID = [Order Details].ProductID)
			                ON Shippers.ShipperID = Orders.ShipVia; "

        Dim cn As String = GetConnectionString()
        Dim da As OleDbDataAdapter = New OleDbDataAdapter(rs, cn)
        Dim dt As DataTable = New DataTable()
        da.Fill(dt)
        Return dt
    End Function

    Private Shared Function GetConnectionString() As String
        Dim path As String = Environment.GetFolderPath(Environment.SpecialFolder.Personal) & "\ComponentOne Samples\Common"
        Dim conn As String = "provider=microsoft.jet.oledb.4.0;data source={0}\c1nwind.mdb;"
        Return String.Format(conn, path)
    End Function
End Class

File: NetFramework/FlexGrid/CS/ColumnBandsOrderManagment/ColumnBandsUserControl.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.OleDb;
using System.Windows.Forms;
using C1.Win.FlexGrid;
using C1.Win.Themes;

namespace ColumnBandsOrderManagment
{
    public enum LayoutOption
    {
        Traditional,
        Compact,
        Detailed
    }

    public partial class ColumnBandsUserControl : UserControl
    {
        public ColumnBandsUserControl()
        {
            InitializeComponent();
            SetUpColSpan();
        }

        /// <summary>
        /// Initializing the control.
        /// </summary>
        public void Init()
        {
            SetupC1FlexGrid();
        }

        /// <summary>
        /// Applyes C1Theme to the user control.
        /// </summary>
        public void ApplyTheme(string themeName)
        {
            var theme = C1ThemeController.GetThemeByName(themeName, false);
            C1ThemeController.ApplyThemeToControlTree(this, theme);
        }

        /// <summary>
        /// Setup C1FlexGrid with data.
        /// </summary>
        private void SetupC1FlexGrid()
        {
            flexGrid1.BeginUpdate();
            flexGrid1.DataSource = GetDataSource();
            flexGrid1.AutoSizeCols();
            flexGrid1.AutoSizeRows();
            flexGrid1.GroupDescriptions = new List<GroupDescription>() { new GroupDescription("CompanyName") };
            flexGrid1.Cols[1].Width = 300;
            flexGrid1.EndUpdate();
            
        }

        /// <summary>
        /// Setup the colSpan property for ColumnBandView
        /// </summary>
        private void SetUpColSpan()
        {
            band2.ColSpan = 1;
            band5.ColSpan = 5;
        }

        #region DataSource
        private DataTable GetDataSource()
        {
            string rs = @"
                SELECT Distinct 
	                Orders.[OrderID],
                    Products.ProductID,
	                Products.ProductName, 
	                Orders.OrderDate, 
                    Orders.ShippedDate as DeliveryDate,
                    Orders.ShipCountry,
                    Orders.ShipCity,
                    Orders.ShipAddress,
	                Customers.CompanyName, 
	                [Order Details].UnitPrice AS UnitPrice, 
	                [Order Details].Quantity, 
	                [Order Details].Discount, 
	                ([Order Details].[UnitPrice]*[Quantity]*(1-[Discount])/100)*100 as Amount 
	                FROM Shippers 
		                INNER JOIN (Products 
		                INNER JOIN ((Employees 
		                INNER JOIN (Customers 
		                INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID) ON Employees.EmployeeID = Orders.EmployeeID) 
		                INNER JOIN [Order Details] ON Orders.OrderID = [Order Details].OrderID) ON Products.ProductID = [Order Details].ProductID)
			                ON Shippers.ShipperID = Orders.ShipVia; ";

            string cn = GetConnectionString();
            OleDbDataAdapter da = new OleDbDataAdapter(rs, cn);
            DataTable dt = new DataTable();
            da.Fill(dt);
            return dt;
        }

        static string GetConnectionString()
        {
            string path = Environment.GetFolderPath(Environment.SpecialFolder.Personal) + @"\ComponentOne Samples\Common";
            string conn = @"provider=microsoft.jet.oledb.4.0;data source={0}\c1nwind.mdb;";
            return string.Format(conn, path);
        }
        #endregion
    }
}